packages/backend-common/src/reading/GithubUrlReader.ts

```diff
@@ -229,6 +234,92 @@ export class GithubUrlReader implements UrlReader {
     throw new Error(message);
   }
 
+  private async getRepositoryArchive(
+    repoUrl: string,
+    branchName: string,
+  ): Promise<Response> {
+    return fetch(new URL(`${repoUrl}/archive/${branchName}.tar.gz`).toString());
```

Comment:
this extra pass through the URL constructor is to catch url format errors i guess?

Reply:
Yeah don't think it does much 😅